modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/IgniteSqlFunctions.java

```diff
@@ -209,6 +216,39 @@ public static BigDecimal toBigDecimal(Object o, int precision, int scale) {
                : toBigDecimal(o.toString(), precision, scale);
     }
 
+    /**
+     * Converts the given {@code BigDecimal} to a decimal with the given {@code precision} and {@code scale}
+     * according to SQL spec for CAST specification: General Rules, 8.
+     */
+    public static BigDecimal convertDecimal(BigDecimal value, int precision, int scale) {
+        assert precision > 0 : "Invalid precision: " + precision;
+
+        int defaultPrecision = IgniteTypeSystem.INSTANCE.getDefaultPrecision(SqlTypeName.DECIMAL);
+        if (precision == defaultPrecision) {
+            // This branch covers at least one known case: access to dynamic parameter from context.
+            // In this scenario precision = DefaultTypePrecision, because types for dynamic params
+            // are created by toSql(createType(param.class)).
+            return value;
+        }
+
+        boolean nonZero = !value.unscaledValue().equals(BigInteger.ZERO);
+
+        if (nonZero) {
+            if (scale > precision) {
+                throw new SqlException(QUERY_INVALID_ERR, "Numeric overflow");
```

Comment:
probably "Numeric overflow" must be closer to PG ? i.e. "Numeric field overflow" ? 
additionally why this message is not a public constant for purpose of further test usage ?

Reply:
Changed a message to `Numeric field overflow`. Moved it to constant but I do not think is a good idea to use the same constant in tests and this method. If we use the same constant, then someone unintentionally changes the constant in `IgniteFunctions` it also changes the tests, thus changing public API. If we use separate constant for tests and system code , then unintentional changes would break tests because error messages won't match.